lib/services/osmdata.dart

```diff
@@ -388,14 +394,17 @@ class OsmData{
       }else{ //route is already long enough, just go back
         routeBack.addAll(graph.AStar(lastVisited, startNode));
       }
-      totalRouteLength += routeBack.map((edge) => edge.weight).fold(0.0, (curr, next) => curr + next);
+      totalRouteLength += lengthOfEdgesKM(routeBack) ;
       route.addAll(routeBack);
       List<Node> routeNodes = List();
       route.forEach((edge) => routeNodes.addAll(graph.edgeToNodes(edge)));
-      result.add(routeNodes);
-      print("TOTALE STRECKENLÄNGE: " + totalRouteLength.toString());
+      results.add(HikingRoute(routeNodes, totalRouteLength, includedPois));
     }
-    return result;
+    return results;
+  }
+
+  double lengthOfEdgesKM(List<Edge> edges){
```

Comment:
lengthOfEdgesKM:
Do we need the accuracy of a doubles or is float enough? If we don't want to support units other than kilometres, we could omit the 'KM' in the method name.

Reply:
Dart only has types num, int and double. Float does not exist as a data type